meta-cli/src/cli/list.rs

```diff
@@ -0,0 +1,175 @@
+
+        match report.stop_reason {
+            StopReason::Error => bail!("Task manager stopped due to error."),
+            StopReason::Manual | StopReason::ManualForced => {
+                bail!("Task manager stopped manually.")
+            }
+            StopReason::Natural => {}
+            StopReason::Restart => panic!("Restart is not supported for serialization."),
+        }
+
+        let typegraphs = report.into_typegraphs()?;
+
+        self.display_typegraphs(dir, typegraphs).await
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct TypegraphObject {
+    name: String,
+    url: String,
+}
+
+impl List {
+    async fn get_more_info(&self, dir: PathBuf) -> Result<Vec<TypegraphObject>> {
+        let query = r#"
+            query {
+                typegraphs {
+                    name
+                    url
+                }
+            }
+        "#;
+        let mut node_config = NodeConfig::default();
+        node_config.url = Url::parse("http://localhost:7891")?;
+        let node = node_config.build(dir).await?;
+        let response = node
+            .post("/typegate")
+            .unwrap()
+            .gql(query.into(), None)
+            .await?;
+
+        response
+            .data("typegraphs")
+            .map_err(|err| anyhow::anyhow!(err))
+    }
+
+    #[allow(clippy::vec_box)]
+    async fn display_typegraphs(
+        &self,
+        dir: PathBuf,
+        typegraphs: Vec<Box<Typegraph>>,
+    ) -> Result<()> {
+        let mut tables: Vec<Table> = Vec::new();
+
+        Table::print_table_header();
+
+        for tg in typegraphs {
+            let mut table = Table::new(tg.name().unwrap());
+            table.set_path(tg.get_path().ok());
+            tables.push(table);
+        }
+
+        if let Ok(more_info) = self.get_more_info(dir).await {
+            for tg in more_info {
+                let mut table = Table::new(tg.name);
+                table.set_url(Some(tg.url));
+                tables.push(table);
+            }
+        }
+
+        Table::print_table_header();
+        for table in tables {
+            println!("{}", table.to_table_row());
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Table {
+    name: String,
+    path: Option<String>,
+    url: Option<String>,
+}
+
+impl Table {
+    fn new(name: String) -> Self {
+        Self {
+            name,
+            path: None,
+            url: None,
+        }
+    }
+
+    fn set_path(&mut self, path: Option<String>) {
+        self.path = path;
+    }
+
+    fn set_url(&mut self, url: Option<String>) {
+        self.url = url;
+    }
+
+    fn to_table_row(&self) -> String {
+        format!(
+            "{:<50} {:<20} {:<30}",
+            self.name,
+            self.path.as_deref().unwrap_or("None"),
+            self.url.as_deref().unwrap_or("None")
+        )
+    }
+
+    fn print_table_header() {
+        println!("{:<50} {:<20} {:<30}", "NAME", "PATH", "URL");
```

Comment:
Some extensions (as this is your first exploratory PR, later tasks shall be fully described from the start) that may be worth implementing :
- add a column to specific which under which target the typography is deployed
- the `local` target could refer to the dev instance
- the target could be `-` for a non deployed tg